google-mail-workitem/src/main/java/org/jbpm/workitem/google/mail/SendMailWorkitemHandler.java

```diff
@@ -0,0 +1,196 @@
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+import javax.activation.DataHandler;
+import javax.activation.DataSource;
+import javax.mail.MessagingException;
+import javax.mail.Multipart;
+import javax.mail.Session;
+import javax.mail.internet.InternetAddress;
+import javax.mail.internet.MimeBodyPart;
+import javax.mail.internet.MimeMessage;
+import javax.mail.internet.MimeMultipart;
+
+import com.google.api.client.util.Base64;
+import com.google.api.services.gmail.Gmail;
+import com.google.api.services.gmail.model.Message;
+import org.jbpm.document.Document;
+import org.jbpm.process.workitem.core.AbstractLogOrThrowWorkItemHandler;
+import org.jbpm.process.workitem.core.util.Wid;
+import org.jbpm.process.workitem.core.util.WidMavenDepends;
+import org.jbpm.process.workitem.core.util.WidParameter;
+import org.jbpm.process.workitem.core.util.WidResult;
+import org.kie.api.runtime.process.WorkItem;
+import org.kie.api.runtime.process.WorkItemManager;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Wid(widfile = "GoogleSendMailDefinitions.wid", name = "GoogleSendMail",
+        displayName = "GoogleSendMail",
+        defaultHandler = "mvel: new org.jbpm.process.workitem.google.mail.SendMailWorkitemHandler()",
+        parameters = {
+                @WidParameter(name = "AppName"),
+                @WidParameter(name = "ClientSecret"),
+                @WidParameter(name = "To"),
+                @WidParameter(name = "From"),
+                @WidParameter(name = "Subject"),
+                @WidParameter(name = "BodyText"),
+                @WidParameter(name = "Attachment")
+        },
+        results = {
+                @WidResult(name = "Message")
+        },
+        mavenDepends = {
+                @WidMavenDepends(group = "com.google.api-client", artifact = "google-api-client", version = "1.23.0"),
+                @WidMavenDepends(group = "com.google.oauth-client", artifact = "google-oauth-client-jetty", version = "1.23.0"),
+                @WidMavenDepends(group = "com.google.apis", artifact = "google-api-services-gmail", version = "v1-rev72-1.23.0")
+        })
+public class SendMailWorkitemHandler extends AbstractLogOrThrowWorkItemHandler {
+
+    private static final Logger logger = LoggerFactory.getLogger(SendMailWorkitemHandler.class);
+    private static final String RESULTS_MESSAGE = "Message";
+
+    private GoogleMailAuth auth = new GoogleMailAuth();
+
+    public void executeWorkItem(WorkItem workItem,
+                                WorkItemManager workItemManager) {
+        Map<String, Object> results = new HashMap<String, Object>();
+        String paramAppName = (String) workItem.getParameter("AppName");
+        String paramClientSecretJSON = (String) workItem.getParameter("ClientSecret");
```

Comment:
wouldn't this be better (app name and client secret) to be given as constructor arguments - making it more a configuration rather than process definition thing?